doc/releases/changelog-dev.md

```diff
@@ -12,11 +12,29 @@
   gate-set that can be translated to the MBQC formalism.
   [(7271)](https://github.com/PennyLaneAI/pennylane/pull/7271)
 
+* Added `convert_to_su2` and `convert_to_su4` to `qml.math` that converts a unitary matrix to SU(2)
+  and SU(4) respectively, and optionally a global phase.
+  [(#7211)](https://github.com/PennyLaneAI/pennylane/pull/7211)
+
+<h4>Resource-efficient Decompositions 🔎</h4>
+
+* The new graph-based decomposition system now supports `QubitUnitary`.
+  [(#7211)](https://github.com/PennyLaneAI/pennylane/pull/7211)
+
+* Added `qml.decomposition.DecompositionNotApplicable` that can be raised from decomposition rules
+  if the decomposition rule is not applicable under a certain set of parameters.
+  [(#7211)](https://github.com/PennyLaneAI/pennylane/pull/7211)
```

Comment:
@isaacdevlugt let me know if we decide on a different API for conditionally applicable decomposition rules.

Reply:
Thanks @astralcai! I think this is fine for now... it's a little clunky looking, but if we have good examples to supplement it with, this will help!